lib/csv.rb

```diff
@@ -1782,9 +1826,19 @@ def lineno
     end
   end
 
-  #
-  # The last row read from this file.
-  #
+  # Returns the row most recently parsed:
+  #   string = "foo,0\nbar,1\nbaz,2\n"
+  #   path = 't.csv'
+  #   File.write(path, string)
+  #   CSV.open(path) do |csv|
+  #     csv.each do |row|
+  #       p [csv.lineno, csv.line]
+  #     end
+  #   end
+  # Output:
+  #   [1, "foo,0\r\n"]
```

Comment:
Not sure if this is happening on a different operating system, but most of the time it should not add `\r` to the  lines read :
```ruby
[1, "foo,0\n"]
[2, "bar,1\n"]
[3, "baz,2\n"]
```

Reply:
This is what I got directly from +irb+, which I am loath to disturb.  No doubt OS-dependent.

Reply:
Did you run it on Windows, right?
Could you use `File.binwrite` instead of `File.write` to avoid adding `\r` automatically on Windows?

Reply:
Or just take out the \r?

Reply:
@kou, willing to remove on your say-so.

Reply:
Removing the `\r` from the output is also OK.

Reply:
Fixed in next push.

Reply:
I have 6 more \r\n, having to do with row_sep. Remove them, too?

Reply:
@kou, thoughts?

Reply:
Yes.

Reply:
Fixed in next push.